### ISSUE TITLE
Idea: Zoom change, when search by radius, polygon or in the current view

### ISSUE THREAD

**Issue by bitmapbulgaria** [hajkmap/Hajk#1162]
When user performing search by polygon, radius or in current map view, the result is zoomed out to whole map. 
action:
![action](https://user-images.githubusercontent.com/57766493/188827027-2a5b30a5-573c-4559-b9f8-8cd1812d5517.png)

result:
![result](https://user-images.githubusercontent.com/57766493/188827135-915a7057-0a88-480b-b6e4-8b558b6de6b2.png)

**Solution**
Would be great the getting the result to change zoom to the "union" of the results in the map. Or may be a button "Zoom to result" to be added in the information window (best is the button effect to zoom to the current level of search results - for example - if you have included in search radius districts, buildings and addresses, you will have three separate lines in the initial result window: there will be nice to have "Zoom to" button, for example. 
Zoom to result:
![search to result](https://user-images.githubusercontent.com/57766493/188827217-4b5fa42d-5634-4a86-afd4-2a429fdf943b.png)

**Alternatives**
Maybe a workaround is not to change zoom, when the search is performed at all (have some disadvantages, though - if an user use polygon for searching and during the polygon creation pans the map, but still).

**Comment by jacobwod** [comment]
I like the suggestion too.

Regarding @Hallbergs answer, I'd like to add that as of [yesterday](https://github.com/hajkmap/Hajk/issues/1159#issuecomment-1238044709) it is now possible to pre-select the desired geographic spatial filter (`within` or `intersects`) on a per-map basis. (You would need to make your own build from `develop` to get this, as it is an upcoming feature in 3.11.) 

This is, of course, an entirely different solution than shutting off the auto zoom and it changes radically how spatial search behaves, but still worth noting I think.

---

On a side note: it's impressive how you managed to navigate around this application. I understand that finding the right switch in Admin can't be easy with everything being written in Swedish. We could really use some help translating this, perhaps dust off this [old branch](https://github.com/hajkmap/Hajk/tree/feature/502-internationalization).

**Comment by bitmapbulgaria** [comment]
Thanks, guys, but my instance's behavior, even with this setting checked is to go to full extend of the map.
Maybe you have updated the source code (my instance is about a two-three weeks old), I'll try to dig the source code, to find the differences. I cannot afford automatic update, because I have translated all menus, alert messages (even statement switches, like on the example) in Bulgarian language and it was a lot work, though. 
P.S. I'm ready to cooperate for translating the system in Bulgarian. Your solution is one of the best WebGIS, I have ever seen. I'm a beginner in JS, Express and React, but even from my point of view, your system and code are impressive- simple, well commented, functional and (especially this) with excellent design. So this is at least, what I can do for Hajk. Maybe not bad idea to prepare a .json with initial strings in Swedish. I'll translate them.

**Comment by Hallbergs** [comment]
Hmm, that sounds strange.. I'll have a look at what might be happening for you when i get a chance. 

I'm glad that you like the solution! We've started the work with internationalizing in [this branch](https://github.com/hajkmap/Hajk/tree/feature/502-internationalization). The internationalization should be on the top of our to-do, just having the app in Swedish is a major limitation. If/when we pick up the internationalizing-project, we'll let you know (so that you can add the Bulgarian translation files :))

**Comment by bitmapbulgaria** [comment]
Update: I have found, that the problem is triggered, when the service tries to draw the search result markers.  
If the option "Rita ut alla sökträffar i kartan automatiskt" in Admin panel is off, the zoom stays the initial zoom of the searching.  Even with this option turned off, after that - when the user click on the results in information window, the service recall the drawing function for the search results markers, and the zoom resets.  The intersect/within choice is not making any change of this behavior (on my instance).
Hope this will help to fix it in the later releases. Will be very grateful, if you give info what file (code part/function) is changed in the fix.

**Comment by jacobwod** [comment]
Yes, you are correct about that. The way we reason is that these are two separate scenarios:
A) User hits "Search". Sometimes we want to narrow down (zoom in or out, depending on the amount of results) the viewport so that only the results are visible. Sometimes we don't want this functionality. So we want a setting (in Admin) where we can turn this on/off.
B) User clicks a result in search results list. We assume that user's action (the "click" action) should lead to a response from the viewport, i.e. user expects something to happen after this click. We zoom in on the selected result. 

I guess (from memory, haven't checked the code this time) that we have a setting for scenario A, but not for B.

Are you suggesting that you would like to disable functionality in scenario B and replace it with a separate "Zoom into result" button?

**Comment by bitmapbulgaria** [comment]
Hi, sorry for the delay. No, I'm not capable to decide "Zoom into result" button or scenario B is better(I'm not yet updated my instance with the new "A" scenario option). Just wanted to share my thoughts about that is causing this full extent behavior.

**Comment by bitmapbulgaria** [comment]
Just another example how wot HAJK instances behave different to same user request:
Example 1 (correct behavior):

https://user-images.githubusercontent.com/57766493/191962047-3b81d2f4-9e33-46fe-8854-ef90e497a122.mp4


Example 2 (wrong behavior, my dev instance, HAJK 3.10):

https://user-images.githubusercontent.com/57766493/191962165-4316f9f9-40c1-42d9-b0c7-83917cd7583e.mp4

Maybe there is some catch, or the Halmstad karta HAJK was updated?

**Comment by Hallbergs** [comment]
Hmm... Do you have your wfs-service open? It would be nice if i could test it out! (You can email your config to [me](mailto:henrik.hallberg@fastighet.goteborg.se) )

**Comment by Hallbergs** [comment]
BTW: We've picked up on the translation work, hopefully we'll have multiple language support very soon!

**Comment by bitmapbulgaria** [comment]
@Hallbergs, I have some difficulties to open the WFS GetFeatureInfo because my Geoserver is behind https NGIX proxy and request crashed, due the mixed content (https and http), don't know why. But I'm sure you can take a look and get the request from my dev instance, on https://geocash.xyz. Just try to search by radius anything in large zoom, to reproduce the error. In the console you can find the requests to the Geoserver, thanks in advance!

**Comment by bitmapbulgaria** [comment]
@Hallbergs , mystery is solved with the strange zoom behavior. If the default map extent is a lot smaller, than the default extent of some of the present layers in the map, then this effect of the zoom change is present. In my case- the few layers had spatial extent of all country territory, not only for the municipality, (the main character in the instance). When I fixed this, all was set and now zoom behavior is as usual. But the idea for 'Zoom to results' is not bad, at all.

**Comment by Hallbergs** [comment]
Ahh i see! Interesting... Im glad that you got it working.
